### DISCUSSION TITLE
feature: add an option to find the slowest tasks

### DISCUSSION THREAD

**Discussion by giancarlosisasi** [vercel/turborepo#7478] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

Hi there
It would be good to have the option to find the top 10 slowest tasks. Similar to how jest-slow-test-reporter works.

With that information, we can focus on reducing the amount of time for that specific tasks. 

My use case is that I have a monorepo with 120 tasks. So it's a little hard to find the slowest tasks just by looking the logs.

### Describe the solution you'd like

A flag in the turbo cli, something like:

`turbo run build --findSlowest=10`

### Describe alternatives you've considered

I have found the `--trace` option but I think this is not what we want to use when trying to find the slowest tasks.

**Comment by NicholasLYang** [comment]
Hi, thanks for opening the issue. One of our goals for Turborepo 1.9 is increased visibility into your tasks, so we'll definitely look into this as a potential feature!

**Comment by giancarlosisasi** [comment]
@NicholasLYang thanks!!!!

**Comment by mehulkar** [comment]
@GiancarlosIO in the latest canary, try running `TURBO_RUN_SUMMARY=true turbo run build` and then looking in `.turbo/runs` directory at the root of your monorepo for a JSON file. It should contain execution timestamps for each task. We hope to make this a nicer experience that you can build on top of. It's unlikely that we'll include something like `--findSlowest` in the 1.9 release, but the raw data might help you get started!

**Comment by mehulkar** [comment]
Update here... you can build this on top of the `--summarize` flag like so:

```bash
turbo build --summarize
SUMMARY_FILE=$(/bin/ls .turbo/runs/*.json | head -n1)
cat $SUMMARY_FILE | jq '[.tasks[] | {"taskId": .taskId, "duration": (.execution.endTime - .execution.startTime)}] | sort_by(-.duration)' 
```

and as plain text output:

```bash
turbo build --summarize
SUMMARY_FILE=$(/bin/ls .turbo/runs/*.json | head -n1)
cat $SUMMARY_FILE | jq -r '[.tasks[] | {"taskId": .taskId, "duration": (.execution.endTime - .execution.startTime)}] | sort_by(-.duration) | .[] | "\(.taskId) \(.duration)"' | column -t
```

This esoteric `jq` usage essentially parses the `execution` field  of every task that was executed, extracts the name and the calculated duration and prints it sorted by duration.

I'm going to keep this open, but hope this helps

**Comment by jeremymeng** [comment]
The workaround is fine but involves finding the file and piping through jq (which may not be installed in CI environments), or some other tools.  It would be nicer if `turbo` has an option to show the durations in the summary
